### PR TITLE
Add Backward compatibility API to C++

### PR DESCRIPTION
**Pull request by muxi**
Add API for user to enable server backward compatibility in C++ server builder.

### CODE REVIEW

File: src/cpp/server/server_builder.cc [hunk 1]
```diff
@@ -358,4 +359,14 @@ void ServerBuilder::InternalAddPluginFactory(
   (*g_plugin_factory_list).push_back(CreatePlugin);
 }
 
+ServerBuilder& ServerBuilder::EnableWorkaround(uint32_t id) {
+  switch (id) {
+    case GRPC_WORKAROUND_ID_CRONET_COMPRESSION:
+      return AddChannelArgument(GRPC_ARG_WORKAROUND_CRONET_COMPRESSION, 1);
+    default:
+      gpr_log(GPR_ERROR, "Workaround %u does not exist or obsolete.", id);
```
Comment (yang-g) [inline src/cpp/server/server_builder.cc]: nit: is obsolete?

**Comment by muxi** [comment]
Test failures:
Asan C - #10653 
Basic Tests Linux - #10653 
Basic Tests Mac - #11116 
BasicTestsWindows - #11188 #9088 
Msan C - #11162 
Portability Tests Linux - #9726
Tsan C - #10894 
gRPC_performance_pull_requests - #11117

File: doc/workarounds.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# gRPC Server Backward Compatibility Issues and Workarounds Manageent
```
Comment (KindDragon) [inline doc/workarounds.md]: Manageent

Comment (muxi) [inline doc/workarounds.md]: Thanks for pointing it out

**Pull request merged**